plugins/timelapse.py

```diff
@@ -267,6 +273,57 @@ def start(self):
 
         dlg.Destroy()
 
+    # Functions to handle the storage of the data in parallel threads
+
+    def _saving_thread(self, i):
+        try:
+            while True:
+                fn, das = self._to_store.get()
+                if fn is None:
+                    self._to_store.task_done()
+                    return
+                logging.info("Saving data %s in thread %d", fn, i)
+                self._exporter.export(fn, das)
+                self._to_store.task_done()
+        except Exception:
+            logging.exception("Failure in the saving thread")
+        finally:
+            logging.debug("Saving thread %d done", i)
+
+    def _start_saving_threads(self, n=4):
+        """
+        n (int >= 1): number of threads
+        """
+        if self._sthreads:
+            logging.warning("The previous saving threads were not stopped, stopping now")
+            self._stop_saving_threads()
+
+        for i in range(n):
+            t = threading.Thread(target=self._saving_thread, args=(i,))
+            t.start()
+            self._sthreads.append(t)
+
+    def _stop_saving_threads(self):
+        """
+        Blocks until all the data has been stored
+        Can be called multiple times in a raw
+        """
+        # Indicate to all the threads that they should stop
+        for t in self._sthreads:
+            self._to_store.put((None, None))  # Special "quit" message
```

Comment:
it feels like you should use the `t` here? Otherwise comment on why this stops all threads in `self._sthreads`.

Reply:
Indeed, it's not obvious. We send one message per thread. We actually just send N messages, and the queue will ensure that each thread gets one.